usr/share/rear/format/USB/default/300_format_usb_disk.sh

```diff
@@ -32,6 +32,25 @@ local current_partition_number=1
 # current start byte of the next partition to add
 local current_partition_start_byte=$(( USB_PARTITION_ALIGN_BLOCK_SIZE * MiB_bytes ))
 
+### Create partition table section
+
+# Initialize USB disk via "parted mklabel" (create partition table)
+# When a format workflow option -b/--bios or -e/--efi was specified set USB_DEVICE_PARTED_LABEL accordingly
+# cf. https://github.com/rear/rear/pull/2828#issuecomment-1164590100
+# and when no format workflow option -b/--bios or -e/--efi was specified
+# then rear/lib/format-workflow.sh sets both FORMAT_BIOS and FORMAT_EFI to 'y'
+# cf. https://github.com/rear/rear/commit/9591fbf77c0c12329738625fcb83bb1d9b419b51
+# to get hybrid boot supporting BIOS and UEFI from the same medium by default
+# see https://github.com/rear/rear/pull/2705
+# so the ordering of the two settings below is crucial
+# to ensure a GUID partition table is set up for hybrid boot:
+is_true "$FORMAT_BIOS" && USB_DEVICE_PARTED_LABEL="msdos"
+is_true "$FORMAT_EFI" && USB_DEVICE_PARTED_LABEL="gpt"
```

Comment:
Would it work to move these two settings to a separate file and symlink it also to the `prep` stage, before `prep/USB/Linux-i386/340_find_mbr_bin.sh` ? This would prevent things from going wrong there.

Reply:
Also, I wonder why to write the MBR in `output` ( `USB/Linux-i386/850_make_USB_bootable.sh` ) instead of in `format`. But that would be a bigger change.

Reply:
@pcahyna 
regarding your first point
https://github.com/rear/rear/pull/2829/files#r905946426

As far as I understand the code
FORMAT_BIOS and FORMAT_EFI are only set in the format workflow
i.e. in lib/format-workflow.sh via command line options
of the "rear format" call
```
# find usr/sbin/rear usr/share/rear -type f | xargs egrep 'FORMAT_BIOS.?=|FORMAT_EFI.?='
usr/share/rear/lib/format-workflow.sh:                FORMAT_BIOS=y
usr/share/rear/lib/format-workflow.sh:                FORMAT_EFI=y
usr/share/rear/lib/format-workflow.sh:        FORMAT_BIOS=y
usr/share/rear/lib/format-workflow.sh:        FORMAT_EFI=y
```

So FORMAT_BIOS and FORMAT_EFI are not set for other workflows
which is the root of my problems with how to reasonably
set USB_DEVICE_PARTED_LABEL for other workflows.

I think some USB device autodetection could help to
automatically set USB_DEVICE_PARTED_LABEL correctly
in other workflows (in particular for "rear mkrescue/mkbackup")
but that would be something for ReaR 2.8.

Reply:
Alternatively prep/USB/Linux-i386/340_find_mbr_bin.sh
could be changed to not depend on USB_DEVICE_PARTED_LABEL
but do some autodetection to directly find out what the
actual USB device partitioning type is (in contrast to
relying on some user config variable value).

Reply:
```
# find usr/sbin/rear usr/share/rear -type f | xargs grep 'Partition Table'
```
shows interesting code how to autodetect the partitioning type
so for my USB disk I get
```
# parted -s /dev/sdb unit MiB print
Model: TOSHIBA External USB 3.0 (scsi)
Disk /dev/sdb: 476940MiB
Sector size (logical/physical): 512B/512B
Partition Table: gpt
Disk Flags: 
Number  Start    End       Size      File system  Name     Flags
 1      0.02MiB  8.00MiB   7.98MiB                primary  bios_grub
 2      8.00MiB  1032MiB   1024MiB   fat32        primary  boot, esp
 3      1032MiB  47694MiB  46662MiB  ext3         primary  legacy_boot

# usb_dev_label=$( parted -s /dev/sdb print | grep -E "Partition Table|Disk label" | cut -d ":" -f "2" | tr -d " " )

# echo "'$usb_dev_label'"
'gpt'
```

Reply:
Got it with the code of the write_protection_ids() function:
```
# usb_partition="/dev/disk/by-label/REAR-000"

# usb_disk="$( lsblk -inpo PKNAME "$usb_partition" 2>/dev/null | awk NF | head -n1 )"

# echo "'$usb_disk'"
'/dev/sdb'

# usb_disk_label=$( parted -s $usb_disk print | grep -E "Partition Table|Disk label" | cut -d ":" -f "2" | tr -d " " )

# echo "'$usb_disk_label'"
'gpt'
```

Reply:
> As far as I understand the code
> FORMAT_BIOS and FORMAT_EFI are only set in the format workflow

Sorry, I did not realize this.

> Alternatively prep/USB/Linux-i386/340_find_mbr_bin.sh
could be changed to not depend on USB_DEVICE_PARTED_LABEL
but do some autodetection to directly find out what the
actual USB device partitioning type is (in contrast to
relying on some user config variable value).

I believe this would be one possible solution, because at this time, the disk already exists, so its partitioning is given. Alternatively, the whole MBR writing stuff could be moved to `format` (MBR could be considered part of the formatting, instead of part of the rescue disk content) and `prep/USB/Linux-i386/340_find_mbr_bin.sh` removed.

Reply:
As far as I see prep/USB/Linux-i386/340_find_mbr_bin.sh
belongs only to SYSLINUX/EXTLINUX as USB disk bootloader
and I won't touch code for SYSLINUX/EXTLINUX because
I know nothing at all about it (except it is a bootloader)
AND
I like to get rid of SYSLINUX/EXTLINUX as USB disk bootloader
and use only GRUB2 when GRUB2 works sufficiently well
but GRUB2 as USB disk bootloader is new in ReaR 2.7
so we have to wait how it behaves out there in practice
until we can think about if an when to drop SYSLINUX/EXTLINUX.
So I won't invest time in SYSLINUX/EXTLINUX improvements.
Only what is needed to keep backward compatibility
as far as possible for us with reasonable effort.